tools/klee/main.cpp

```diff
@@ -72,130 +72,191 @@ namespace {
-               cl::desc("Consider the function with the given name as the entrypoint"),
-               cl::init("main"));
-
-  cl::opt<std::string>
-  RunInDir("run-in", cl::desc("Change to the given directory prior to executing"));
-
-  cl::opt<std::string>
-  Environ("environ", cl::desc("Parse environ from given file (in \"env\" format)"));
-
   cl::list<std::string>
   InputArgv(cl::ConsumeAfter,
             cl::desc("<program arguments>..."));
 
-  cl::opt<bool>
-  NoOutput("no-output",
-           cl::desc("Don't generate test files"));
 
-  cl::opt<bool>
-  WarnAllExternals("warn-all-externals",
-                   cl::desc("Give initial warning for all externals."));
+  /*** Test case options ***/
+
+  cl::OptionCategory TestCaseCat("Test case options",
+                                 "These options select the files to generate for each test case.");
 
   cl::opt<bool>
   WriteCVCs("write-cvcs",
-            cl::desc("Write .cvc files for each test case"));
+            cl::desc("Write .cvc files for each test case (default=false)"),
+            cl::cat(TestCaseCat));
 
   cl::opt<bool>
   WriteKQueries("write-kqueries",
-            cl::desc("Write .kquery files for each test case"));
+                cl::desc("Write .kquery files for each test case (default=false)"),
+                cl::cat(TestCaseCat));
 
   cl::opt<bool>
   WriteSMT2s("write-smt2s",
-            cl::desc("Write .smt2 (SMT-LIBv2) files for each test case"));
+             cl::desc("Write .smt2 (SMT-LIBv2) files for each test case (default=false)"),
+             cl::cat(TestCaseCat));
 
   cl::opt<bool>
   WriteCov("write-cov",
-           cl::desc("Write coverage information for each test case"));
+           cl::desc("Write coverage information for each test case (default=false)"),
+           cl::cat(TestCaseCat));
 
   cl::opt<bool>
   WriteTestInfo("write-test-info",
-                cl::desc("Write additional test case information"));
+                cl::desc("Write additional test case information (default=false)"),
+                cl::cat(TestCaseCat));
 
   cl::opt<bool>
   WritePaths("write-paths",
-                cl::desc("Write .path files for each test case"));
+             cl::desc("Write .path files for each test case (default=false)"),
+             cl::cat(TestCaseCat));
 
   cl::opt<bool>
   WriteSymPaths("write-sym-paths",
-                cl::desc("Write .sym.path files for each test case"));
+                cl::desc("Write .sym.path files for each test case (default=false)"),
+                cl::cat(TestCaseCat));
+
+
+  /*** Starting options ***/
+
+  cl::OptionCategory StartCat("Starting options",
+                              "These options affect how execution is started.");
+
+  cl::opt<std::string>
+  EntryPoint("entry-point",
+             cl::desc("Function in which to start execution (default=main)"),
+             cl::init("main"),
+             cl::cat(StartCat));
+
+  cl::opt<std::string>
+  RunInDir("running-dir",
```

Comment:
Maybe `run-dir`? But I still prefer `run-in`...

Reply:
I like `run-in` too. "running a directory" doesn't click with me.

Reply:
I like `run-in-dir` too.

Reply:
I'm fine with it too.